Implemented register with temporary ArrayList.

blankss opened the pull request:
I implemented the registration portion of our product and verified that it worked through Postman. The @PostMapping("/register") is the URL to register a user in which the JSON data will be sent to the backend to store the user information. @RequestBody is basically the data that we are sending into the method. I then used a loop to see if the user exists already in the temporary ArrayList through the equals method that I wrote that mainly compared the emails and usernames of the new user and the existing users. If the user exists already, then the method returns a string that says that a user is already created with that data and cannot be created. If the user does not exist, then we add it to the temporary storage for now. 

I also modified the @GetMapping to display the ArrayList to see the data being stored.

blankss committed:
Implemented register with temporary ArrayList and added equals method in User.

khainl1110 commented:
Looks good. However, make sure you url path follows rest-api design. /users to get all users. /users/{id} to create a user. Read more here: https://www.sitepoint.com/rest-api/. Your current url is more for frontend. In addition, please create PUT and DELETE method as well. For POST and PUT method, please return the created/modified user object for now because we need to return a json data.

blankss committed:
Added DELETE and PUT and also modified mapping to id.

blankss commented:
Okay, I addressed the issues. Added the Put and Delete mapping and modified the mapping to go to /users/{id} for these methods. Researched a bit on @RequestBody and @PathVariable parameters as well to use that id.

khainl1110 committed:
Add some comments for later revision, can merged
